price-estimator/openapi.yml

```diff
@@ -0,0 +1,81 @@
+openapi: 3.0.3
+info:
+  version: 0.0.1
+  title: Price Estimator OpenAPI
+paths:
+  /api/v1/markets/{market}/estimated-buy-amount/{sell amount in quote}:
+    get:
+      summary: Estimated Buy Amount
+      description: Estimate the buy amount (in buy tokens) a user can set as a limit order while still expecting to be completely matched when selling the given amount of quote token.
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/definitions/AmountResponseSchema"
+      parameters:
+        - $ref: "#/definitions/MarketParameter"
+        - name: sell amount in quote
+          required: true
+          in: path
+          schema:
+            $ref: "#/definitions/NumberParameterSchema"
+          example: 1
+        - $ref: "#/definitions/UnitParameter"
+  /api/v1/markets/{market}/estimated-amounts-at-price/{price}:
+    get:
+      summary: Estimated Amounts At Price
+      description: Estimate largest buy and sell amounts for a limit at given price that would get matched. Note that it might be possible to use a higher buy amount for the same returned sell amount and still likely get completely matched by the solver. This buy amount can be computed with a subsequent estimate buy amount API call using the returned sell amount in quote value.
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/definitions/AmountResponseSchema"
+      parameters:
+        - $ref: "#/definitions/MarketParameter"
+        - name: price
+          required: true
+          in: path
+          schema:
+            $ref: "#/definitions/NumberParameterSchema"
+          example: 400
+        - $ref: "#/definitions/UnitParameter"
+definitions:
+  MarketParameter:
+    name: market
+    in: path
+    description: token pair of the form `<buy token>-<sell token>`
+    required: true
+    schema:
+      type: string
+      example: 1-7
+  UnitParameter:
+    name: unit
+    in: query
+    description: If `baseunits` (the default) all amounts are denominated in the "natural" unit of the respective token given by the number of decimals specified through the ERC20 interface. If set to `atoms` all amounts are denominated in the smallest available unit (atom) of the token.
+    required: false
+    schema:
+      type: string
+      enum: [baseunits, atoms]
+      example: baseunits
```

Comment:
I am assuming Nick's PR that changes the unit parameter to work like this instead of `atoms=true|false` gets merged. That PR deprecates the atoms syntax so we shouldn't document it.